sw/airborne/arch/chibios/modules/actuators/esc_dshot.c

```diff
@@ -0,0 +1,482 @@
+ * @note      see also dshotSendThrottles
+ * @api
+ */
+void dshotSetThrottle(DSHOTDriver *driver, const  uint8_t index,
+                      const  uint16_t throttle)
+{
+  //if (throttle == 0) {
+  //  // send motor off
+  //  dshotSendSpecialCommand(driver, index, DSHOT_CMD_MOTOR_STOP);
+  //} else
+  if (throttle > 0 && throttle <= DSHOT_CMD_MAX) {
+    return; // special commands (except MOTOR_STOP) can't be applied from this function
+  } else {
+    // send normal throttle
+    if (index < DSHOT_CHANNELS) {
+      setDshotPacketThrottle(&driver->dshotMotors.dp[index], Min(throttle, DSHOT_MAX_VALUE));
+    } else if (index == DSHOT_ALL_MOTORS) {
+      for (uint8_t _index = 0; _index < DSHOT_CHANNELS; _index++) {
+        setDshotPacketThrottle(&driver->dshotMotors.dp[_index], Min(throttle, DSHOT_MAX_VALUE));
+      }
+    }
+  }
+}
+
+/**
+ * @brief   send special order to one of the ESC (BHELIX, KISS, ...)
+ *
+ * @param[in] driver    pointer to the @p DSHOTDriver object
+ * @param[in] index     channel : [0..3] or [0..1] depending on driver used
+ * @param[in] specmd   special commands, see enum
+ * @api
+ */
+void dshotSendSpecialCommand(DSHOTDriver *driver, const  uint8_t index,
+                             const dshot_special_commands_t specmd)
+{
+  if (specmd > DSHOT_CMD_MAX) {
+    return; // Don't apply special commands from this function
+  }
+  if (index < DSHOT_CHANNELS) {
+    setDshotPacketThrottle(&driver->dshotMotors.dp[index], specmd);
+    setDshotPacketTlm(&driver->dshotMotors.dp[index], driver->config->tlm_sd != NULL);
+  } else if (index == DSHOT_ALL_MOTORS) {
+    for (uint8_t _index = 0; _index < DSHOT_CHANNELS; _index++) {
+      setDshotPacketThrottle(&driver->dshotMotors.dp[_index], specmd);
+      setDshotPacketTlm(&driver->dshotMotors.dp[_index], driver->config->tlm_sd != NULL);
+    }
+  }
+
+  uint8_t repeat;
+  switch (specmd) {
+    case DSHOT_CMD_SPIN_DIRECTION_1:
+    case DSHOT_CMD_SPIN_DIRECTION_2:
+    case DSHOT_CMD_3D_MODE_OFF:
+    case DSHOT_CMD_3D_MODE_ON:
+    case DSHOT_CMD_SAVE_SETTINGS:
+    case DSHOT_CMD_SETTINGS_REQUEST:
+    case DSHOT_CMD_AUDIO_STREAM_MODE_ON_OFF:
+    case DSHOT_CMD_SILENT_MODE_ON_OFF:
+      repeat = 10;
+      break;
+    default:
+      repeat = 1;
+  }
+
+  while (repeat--) {
+    dshotSendFrame(driver);
+    chThdSleepMilliseconds(1);
+  }
+}
+
+/**
+ * @brief   send throttle packed order to all of the ESCs
+ *
+ * @param[in] driver    pointer to the @p DSHOTDriver object
+ * @param[in] throttle[DSHOT_CHANNELS]  [48 .. 2047]
+ * @note dshotSendFrame is called by this function
+ * @note telemetry bit is set in turn for each ESC of the ESCs
+ * @api
+ */
+void dshotSendThrottles(DSHOTDriver *driver, const  uint16_t throttles[DSHOT_CHANNELS])
+{
+  for (uint8_t index = 0; index < DSHOT_CHANNELS; index++) {
+    setDshotPacketThrottle(&driver->dshotMotors.dp[index], throttles[index]);
+  }
+
+  dshotSendFrame(driver);
+}
+
+
+
```

Comment:
A lot of whitespace.. does the fix_code_style fix this?

Reply:
no, only trailing white-spaces